### PR TITLE
Don't silently regenerate wallet

### PR DESCRIPTION
**Pull request by Krigpl**
Those cases shouldn't happen but if they do the solution isn't to quietly generate new identity imo.

### CODE REVIEW

**Comment by etam** [comment]
@badb decided this should be this way for now, but it should be changed before mainnet release. See #2213 

I don't know if just throwing an exception is a good enough solution.

**Comment by Krigpl** [comment]
My reasoning is that this scenario either:
- will not happen in mainnet at all
- will happen in mainnet and handling it should be muuuuuch more carefully done, including informing the user about backup, why the address has changed, why they lost all theirs reputation etc

Second case is obviously much more complicated but we don't plan supporting that at the moment, so we're fine with 1. In which case exceptions are fine. For testnet we can just ask users to uninstall previous golem version before installing new one which we has done in the past iirc.

Anyway, I don't really need this change, I just thought it'd be clearer this way.

File: golem/core/keysauth.py [hunk 1]
```diff
@@ -125,14 +123,13 @@ def _load_and_check_keys(priv_key_path: str,
...
-            return None
+            raise Exception(
+                "Wrong loaded private key size: {}".format(len(priv_key)))
 
         pub_key = privtopub(priv_key)
 
         if not KeysAuth.is_pubkey_difficult(pub_key, difficulty):
-            logger.warning("Loaded key is not difficult enough.")
-            return None
+            raise Exception("Loaded key is not difficult enough")
```
Comment (tworec) [inline golem/core/keysauth.py:132]: I think we should point out some solution, for example to reinstall golem, or delete old key file... or even some url to some FAQ page with question: how to transfer GNT(W) from old golem account...

**Review by tworec — CHANGES_REQUESTED** [review]
@Krigpl your reasoning is ok. I have just one comment

File: golem/core/keysauth.py (hunk 1 above)
Comment (etam) [reply golem/core/keysauth.py:132]: #2213

Comment (tworec) [reply golem/core/keysauth.py:132]: yes, but until #2213 will happened we should, at least, inform user how to deal with this issue... for example "Loaded key is not difficult enough. To regenerate please reinstall golem or remove key file `path`.

Comment (etam) [reply golem/core/keysauth.py:132]: I think this PR should be closed without merging. This issue should be properly and completely handled in #2213 and any partial solutions are bad in their own ways and not worth spending time.

Comment (tworec) [reply golem/core/keysauth.py:132]: I agree the latter part of your statement, but this PR is good foundation to final #2213 solution :)

**Comment by tworec** [comment]
included in #2281

**Pull request closed without merging**